system-tests/test.py

```diff
@@ -138,7 +145,9 @@ def _assert_diff(self, first, second):
 
     sequence = [
         FabTestCase(root_dir / 'MinimalFortran'),
-        FabTestCase(root_dir / 'FortranDependencies')
+        FabTestCase(root_dir / 'FortranDependencies'),
+        FabTestCase(root_dir / 'FortranPreProcess',
+                    fpp_flags='-DSHOULD_I_STAY=yes')
```

Comment:
I agree with the developer that we should probably test the undefined situation as well. This will require additional system testing infrastrucgture so I am happy for it to be postponed to a future ticket provided one is created and linked to this review.

Reply:
I think I will postpone this until after #39 goes on, since that already makes some headway in this regard (and I'd rather not have us stomp all over each other's work)

Opened #42 to deal with this